Src/IronPython/Modules/Builtin.cs

```diff
@@ -925,36 +934,55 @@ public class mapobject : IEnumerable {
         }
 
         public static object max(CodeContext/*!*/ context, object x, object y, [ParamDictionary] IDictionary<object, object> dict) {
-            object method = GetMaxKwArg(dict);
+            var kwargTuple = GetMaxKwArg(dict, isDefaultAllowed: false);
+            object method = kwargTuple.Item1;
+
             PythonContext pc = context.LanguageContext;
             return pc.GreaterThan(PythonCalls.Call(context, method, x), PythonCalls.Call(context, method, y)) ? x : y;
         }
 
         public static object max(CodeContext/*!*/ context, [ParamDictionary]IDictionary<object, object> dict, params object[] args) {
+            var kwargTuple = GetMaxKwArg(dict, isDefaultAllowed: false);
+            object method = kwargTuple.Item1;
+            object def = kwargTuple.Item2;
             if (args.Length > 0) {
                 int retIndex = 0;
                 if (args.Length == 1) {
                     return max(context, args[retIndex], dict);
                 }
-                object method = GetMaxKwArg(dict);
+                
                 object retValue = PythonCalls.Call(context, method, args[retIndex]);
                 PythonContext pc = context.LanguageContext;
                 for (int i = 1; i < args.Length; i++) {
                     object tmpRetValue = PythonCalls.Call(context, method, args[i]);
-                    if (pc.GreaterThan(tmpRetValue, retValue)) {
+                    var tmptype = tmpRetValue.GetType();
+                    var rettype = retValue.GetType();
+
+                    if ( tmptype == typeof(bool) && rettype == typeof(PythonList) ) {
```

Comment:
What are these checks for? Are they to get specific tests passing? If so the checks should be removed and the tests disabled.

I'm not sure if I've filed an issue already, but comparisons between uncomparable types need to be removed. This means that these cases should fail properly without special handling.